### PR TITLE
BLDC code including ramping

### PR DESCRIPTION
**Pull request by BlakeLazarine**
Changes based on previous pull request. Modified ramping to be more generalized. Still to be done: different ramping methods (will not change interface)

### CODE REVIEW

File: motor_control/brushless/GLOBAL.h [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+  const long maxWaitTime = 1500;
+
+  MotorControl(){ timeMark = millis();  }
+
+  void initSensors(){
+    for(int i = 0; i < NUM_PINS; i++)
+      hallSensor[i] = digitalRead(hallPins[i]);
+  }
+
+  bool isMotorStopped(){  return millis() - timeMark >= maxWaitTime;  }
```
Comment (ashvinnagarajan) [inline motor_control/brushless/GLOBAL.h:45]: Will timeMark always have a value before this is called? i see timeMark doesn't get initialized when declared

File: motor_control/brushless/GLOBAL.h [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+  void rotateMotor(){
+    byte index = 4 * (byte)hallSensor[0] + 2 * (byte)hallSensor[1] + (byte)hallSensor[2] - 1;
```
Comment (ashvinnagarajan) [inline motor_control/brushless/GLOBAL.h:48]: Just curious: where does this come from?

File: motor_control/brushless/ramping.cpp [hunk 3]
```diff
@@ -0,0 +1,65 @@
+#include "ramping.h"
+#include "GLOBAL.h"
+Ramping::Ramping():m_currentSpd(MIN_SPD), m_desiredSpd(MIN_SPD){}
+void Ramping::setCurrentSpd(byte spd)      {  m_currentSpd = spd; }
+void Ramping::setDesiredSpd(byte spd)      {  m_desiredSpd = spd; }
+byte Ramping::getCurrentSpd() const {  return m_currentSpd; }
+byte Ramping::getDesiredSpd() const {  return m_desiredSpd; }
+
+bool Ramping::isTimeToChangeSpd(long timeMillis){
```
Comment (ashvinnagarajan) [inline motor_control/brushless/ramping.cpp]: Why pass this parameter if its not used?

**Review by ashvinnagarajan — APPROVED** [review]
Looks great!

**Review by abdullahelqaq — CHANGES_REQUESTED** [review]
The ramping implementation itself looks awesome!

My concern with this ramping class though is that it is essentially creating an alternative throttle implementation than the one in `motor_control/throttle`, which can quickly cause confusion. Is there a reason this class exists specifically under `motor_control/brushless`? It's my understanding that ramping does not depend on the type of motor used.

Is there a way we can reconcile this implementation with the code in `motor_control/throttle`? Even if that involves making significant changes/replacements to that code, I would prefer that over having two concurrent throttle implementations on `master`. Happy to hear other thoughts on this though!

File: motor_control/brushless/GLOBAL.h (hunk 1 above)
Comment (BlakeLazarine) [reply motor_control/brushless/GLOBAL.h:45]: good point. Changed

File: motor_control/brushless/GLOBAL.h (hunk 2 above)
Comment (BlakeLazarine) [reply motor_control/brushless/GLOBAL.h:48]: Essentially we have an array that can be used if needed to map the hall sensor inputs into their corresponding outputs. This index variable is a means of calculating which index corresponds to the current hall sensor inputs.

File: motor_control/brushless/ramping.cpp (hunk 3 above)
Comment (BlakeLazarine) [reply motor_control/brushless/ramping.cpp]: forgot to change the internal function calls lol

**Pull request merged**